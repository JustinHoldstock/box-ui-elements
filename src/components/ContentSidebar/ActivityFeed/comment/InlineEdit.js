/**
 * @flow
 * @file Inline Edit component
 */

import React, { Component } from 'react';
import { injectIntl, intlShape } from 'react-intl';

import PlainButton from 'box-react-ui/lib/components/plain-button';
import IconPencil from 'box-react-ui/lib/icons/general/IconPencil';

import messages from '../messages';

type Props = {
    toEdit: Function,
    id: Function,
    intl: intlShape.isRequired
};

class InlineEdit extends Component<Props> {
    onEdit = () => {
        const { id, toEdit } = this.props;
        toEdit({ id });
    };

    render() {
        const { onEdit } = this;
        return (
            <div className='box-ui-comment-edit-container'>
                <PlainButton
                    aria-label={this.props.intl.formatMessage(messages.editLabel)}
                    className='box-ui-comment-edit'
                    onClick={onEdit}
                    type='button'
                >
                    <IconPencil />
                </PlainButton>
            </div>
        );
    }
}

export { InlineEdit as InlineEditBase };
export default injectIntl(InlineEdit);
