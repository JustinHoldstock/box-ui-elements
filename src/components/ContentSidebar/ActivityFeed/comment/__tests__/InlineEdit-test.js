import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';

import { InlineEditBase as InlineEdit } from '../InlineEdit';

const sandbox = sinon.sandbox.create();

describe('features/activity-feed/comment/InlineEdit', () => {
    const intl = { formatMessage: () => {} };

    afterEach(() => {
        sandbox.verifyAndRestore();
    });

    test('should correctly render comment', () => {
        const wrapper = shallow(<InlineEdit id='123' intl={intl} toEdit={() => {}} />);

        expect(wrapper.hasClass('box-ui-comment-edit-container')).toBe(true);
    });

    test('should call toEdit handler when comment deletion is confirmed', () => {
        const toEditSpy = sandbox.spy();

        const wrapper = mount(<InlineEdit id='123' intl={intl} toEdit={toEditSpy} />);

        const editBtn = wrapper.find('.box-ui-comment-edit').hostNodes();
        editBtn.simulate('click');

        expect(toEditSpy.calledWith({ id: '123' })).toBe(true);
    });
});
