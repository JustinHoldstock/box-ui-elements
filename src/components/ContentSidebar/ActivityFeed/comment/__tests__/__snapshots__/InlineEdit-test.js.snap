// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/activity-feed/comment/InlineEdit should correctly render comment 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InlineEdit
    id="123"
    intl={
      Object {
        "formatMessage": [Function],
      }
    }
    toEdit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <PlainButton
        aria-label={undefined}
        className="box-ui-comment-edit"
        onClick={[Function]}
        type="button"
      >
        <IconPencil />
      </PlainButton>,
      "className": "box-ui-comment-edit-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "aria-label": undefined,
        "children": <IconPencil />,
        "className": "box-ui-comment-edit",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <PlainButton
          aria-label={undefined}
          className="box-ui-comment-edit"
          onClick={[Function]}
          type="button"
        >
          <IconPencil />
        </PlainButton>,
        "className": "box-ui-comment-edit-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "aria-label": undefined,
          "children": <IconPencil />,
          "className": "box-ui-comment-edit",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
